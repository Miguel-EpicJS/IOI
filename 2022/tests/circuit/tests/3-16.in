385045b28fbc1903eca997d513040235e0229e01
1000 1000 5
-1 0 1 2 3 4 5 6 7 8 9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 53 54 55 56 57 58 59 60 61 62 63 64 65 66 67 68 69 70 71 72 73 74 75 76 77 78 79 80 81 82 83 84 85 86 87 88 89 90 91 92 93 94 95 96 97 98 99 100 101 102 103 104 105 106 107 108 109 110 111 112 113 114 115 116 117 118 119 120 121 122 123 124 125 126 127 128 129 130 131 132 133 134 135 136 137 138 139 140 141 142 143 144 145 146 147 148 149 150 151 152 153 154 155 156 157 158 159 160 161 162 163 164 165 166 167 168 169 170 171 172 173 174 175 176 177 178 179 180 181 182 183 184 185 186 187 188 189 190 191 192 193 194 195 196 197 198 199 200 201 202 203 204 205 206 207 208 209 210 211 212 213 214 215 216 217 218 219 220 221 222 223 224 225 226 227 228 229 230 231 232 233 234 235 236 237 238 239 240 241 242 243 244 245 246 247 248 249 250 251 252 253 254 255 256 257 258 259 260 261 262 263 264 265 266 267 268 269 270 271 272 273 274 275 276 277 278 279 280 281 282 283 284 285 286 287 288 289 290 291 292 293 294 295 296 297 298 299 300 301 302 303 304 305 306 307 308 309 310 311 312 313 314 315 316 317 318 319 320 321 322 323 324 325 326 327 328 329 330 331 332 333 334 335 336 337 338 339 340 341 342 343 344 345 346 347 348 349 350 351 352 353 354 355 356 357 358 359 360 361 362 363 364 365 366 367 368 369 370 371 372 373 374 375 376 377 378 379 380 381 382 383 384 385 386 387 388 389 390 391 392 393 394 395 396 397 398 399 400 401 402 403 404 405 406 407 408 409 410 411 412 413 414 415 416 417 418 419 420 421 422 423 424 425 426 427 428 429 430 431 432 433 434 435 436 437 438 439 440 441 442 443 444 445 446 447 448 449 450 451 452 453 454 455 456 457 458 459 460 461 462 463 464 465 466 467 468 469 470 471 472 473 474 475 476 477 478 479 480 481 482 483 484 485 486 487 488 489 490 491 492 493 494 495 496 497 498 499 500 501 502 503 504 505 506 507 508 509 510 511 512 513 514 515 516 517 518 519 520 521 522 523 524 525 526 527 528 529 530 531 532 533 534 535 536 537 538 539 540 541 542 543 544 545 546 547 548 549 550 551 552 553 554 555 556 557 558 559 560 561 562 563 564 565 566 567 568 569 570 571 572 573 574 575 576 577 578 579 580 581 582 583 584 585 586 587 588 589 590 591 592 593 594 595 596 597 598 599 600 601 602 603 604 605 606 607 608 609 610 611 612 613 614 615 616 617 618 619 620 621 622 623 624 625 626 627 628 629 630 631 632 633 634 635 636 637 638 639 640 641 642 643 644 645 646 647 648 649 650 651 652 653 654 655 656 657 658 659 660 661 662 663 664 665 666 667 668 669 670 671 672 673 674 675 676 677 678 679 680 681 682 683 684 685 686 687 688 689 690 691 692 693 694 695 696 697 698 699 700 701 702 703 704 705 706 707 708 709 710 711 712 713 714 715 716 717 718 719 720 721 722 723 724 725 726 727 728 729 730 731 732 733 734 735 736 737 738 739 740 741 742 743 744 745 746 747 748 749 750 751 752 753 754 755 756 757 758 759 760 761 762 763 764 765 766 767 768 769 770 771 772 773 774 775 776 777 778 779 780 781 782 783 784 785 786 787 788 789 790 791 792 793 794 795 796 797 798 799 800 801 802 803 804 805 806 807 808 809 810 811 812 813 814 815 816 817 818 819 820 821 822 823 824 825 826 827 828 829 830 831 832 833 834 835 836 837 838 839 840 841 842 843 844 845 846 847 848 849 850 851 852 853 854 855 856 857 858 859 860 861 862 863 864 865 866 867 868 869 870 871 872 873 874 875 876 877 878 879 880 881 882 883 884 885 886 887 888 889 890 891 892 893 894 895 896 897 898 899 900 901 902 903 904 905 906 907 908 909 910 911 912 913 914 915 916 917 918 919 920 921 922 923 924 925 926 927 928 929 930 931 932 933 934 935 936 937 938 939 940 941 942 943 944 945 946 947 948 949 950 951 952 953 954 955 956 957 958 959 960 961 962 963 964 965 966 967 968 969 970 971 972 973 974 975 976 977 978 979 980 981 982 983 984 985 986 987 988 989 990 991 992 993 994 995 996 997 998 0 1 2 4 5 6 3 7 8 10 9 11 12 13 16 15 17 14 19 18 21 20 23 22 24 25 27 26 30 28 29 31 33 32 34 35 36 38 37 40 42 43 39 41 45 44 48 47 49 52 53 46 54 51 55 50 58 56 60 57 63 62 64 59 65 61 66 67 68 71 69 70 72 73 74 77 75 76 78 79 80 81 85 83 82 84 86 87 88 89 90 91 92 93 94 95 96 97 98 99 101 102 100 104 105 103 107 108 110 111 109 106 112 114 113 115 116 117 118 119 121 120 122 123 124 125 126 129 130 132 133 131 128 127 134 135 137 136 140 141 139 142 138 143 144 145 146 148 149 147 150 151 152 153 155 156 154 157 158 160 159 161 162 163 164 166 165 168 169 167 170 174 172 175 176 173 177 171 178 179 181 180 183 182 184 185 186 187 188 189 190 191 192 193 195 194 196 198 197 199 200 201 203 202 204 206 205 207 208 210 209 212 211 214 215 216 213 219 218 217 220 221 223 225 222 226 227 224 228 229 231 230 232 234 233 236 237 238 235 241 240 239 242 244 245 243 247 248 246 250 252 249 251 253 254 255 256 257 258 259 260 261 263 265 264 262 267 266 269 268 271 270 273 275 272 276 274 277 278 279 280 281 283 284 282 285 286 287 288 290 289 292 291 293 295 296 294 297 298 299 300 301 304 305 302 303 307 309 306 308 311 310 314 312 313 315 316 317 318 320 319 321 322 323 324 325 326 327 328 329 330 332 331 333 334 335 336 338 340 339 337 341 342 344 343 345 346 347 348 350 351 349 353 354 352 355 356 358 357 359 360 362 363 365 361 367 368 369 366 364 371 372 373 370 374 375 376 377 379 378 381 380 382 383 384 385 388 386 389 387 390 391 392 393 395 396 397 394 399 398 401 403 402 400 404 406 407 405 408 409 410 411 412 414 413 417 415 418 419 420 423 416 424 421 422 425 426 427 428 429 430 431 432 433 435 436 434 437 438 441 440 442 443 444 439 445 446 447 449 448 451 450 452 453 454 455 456 457 458 459 460 461 462 464 466 467 463 468 465 470 472 469 473 471 474 475 476 477 478 479 481 480 482 483 484 486 485 487 488 489 491 490 492 494 495 493 496 497 498 499 501 503 502 500 505 506 504 507 508 509 511 510 512 514 513 516 515 520 518 517 521 523 519 522 525 524 528 527 526 529 530 531 532 533 534 536 535 537 538 540 539 541 542 543 544 545 546 547 548 549 550 551 552 554 553 556 558 559 555 557 560 561 562 563 564 565 567 566 568 569 570 572 573 574 576 577 578 575 571 579 581 580 582 584 585 583 588 587 589 586 591 592 593 596 590 597 594 598 595 601 600 599 603 605 604 606 607 602 609 608 610 611 613 614 612 615 617 616 620 618 619 622 621 623 624 625 627 626 629 628 632 631 630 633 634 635 636 637 638 639 640 641 642 644 643 647 646 648 649 645 650 651 652 654 653 655 656 657 658 659 660 661 663 662 664 666 665 667 668 669 670 672 671 674 675 673 676 677 678 679 680 682 681 683 687 686 684 685 688 689 690 692 691 696 695 694 697 693 698 699 700 702 701 703 705 706 707 704 708 709 712 710 713 714 711 715 716 717 718 719 721 720 722 723 724 726 725 727 728 729 730 731 733 732 735 734 736 737 739 738 743 741 742 740 744 745 746 748 749 747 751 750 752 753 754 758 756 757 755 759 761 760 762 763 764 765 768 766 769 767 770 771 773 774 772 777 775 776 778 779 780 781 782 784 783 785 787 786 790 792 789 791 788 793 794 795 798 799 796 801 802 797 803 800 804 806 807 809 808 811 810 813 814 815 805 812 816 817 818 819 822 820 824 821 828 829 827 825 823 830 831 833 826 834 832 836 835 837 839 838 841 840 842 843 844 848 846 847 849 845 851 850 852 854 853 855 856 857 858 859 861 860 863 862 866 864 865 869 868 870 867 872 873 874 871 875 876 877 878 879 880 881 884 882 885 886 883 887 891 892 890 888 889 893 894 895 896 897 898 899 901 900 903 902 904 905 906 907 908 909 910 911 912 913 914 915 919 918 920 916 917 922 923 921 924 925 926 927 928 929 930 931 933 934 932 936 937 940 939 938 935 941 943 942 945 944 946 948 949 947 950 951 952 953 954 955 956 957 958 959 960 962 963 964 965 961 966 967 968 969 970 971 974 973 972 975 977 976 978 979 980 984 983 981 985 986 987 982 988 989 991 990 992 993 994 995 999 998 996 997
0 0 1 0 0 1 0 0 0 0 1 1 1 1 1 1 1 1 0 1 1 0 0 1 0 0 0 0 1 1 0 0 0 0 0 1 0 0 0 1 0 0 1 0 0 0 1 0 1 1 0 1 1 0 0 1 0 0 1 1 0 0 0 0 0 0 1 0 0 0 0 0 0 0 0 1 1 0 1 0 0 0 1 0 0 0 0 1 0 1 1 1 1 0 1 1 1 1 1 1 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 0 1 0 0 1 1 0 0 0 1 1 0 0 1 0 0 1 1 0 1 1 0 1 0 1 0 1 0 1 0 1 1 1 1 0 1 1 1 0 0 1 1 0 0 0 0 0 0 1 1 1 0 0 0 0 0 1 1 0 1 1 0 1 1 1 1 1 1 1 1 0 1 0 0 0 0 0 1 1 1 1 1 1 0 0 0 0 0 1 0 0 1 1 1 0 1 1 0 0 0 1 1 0 1 1 0 1 1 0 0 1 1 1 1 1 1 0 0 0 1 1 0 1 0 1 1 1 0 1 0 0 1 0 1 0 1 0 0 1 1 1 1 0 0 1 1 1 0 0 0 1 1 1 0 1 0 1 0 0 1 1 1 0 1 1 1 1 1 1 0 0 1 1 1 0 0 1 0 0 1 1 1 1 1 1 0 1 0 1 0 1 0 0 1 1 0 0 1 0 1 1 1 1 0 0 0 1 1 0 1 1 0 1 0 1 0 0 1 0 0 0 0 0 1 0 1 1 1 1 1 1 1 1 1 1 0 1 1 0 0 0 1 0 0 0 0 1 0 0 1 1 0 1 0 1 0 0 0 0 1 0 0 1 1 0 0 1 0 1 1 1 0 0 0 0 1 0 1 1 0 1 1 1 1 1 0 1 1 1 1 1 1 1 0 1 0 0 0 1 1 1 0 1 0 0 0 0 0 0 1 0 0 1 0 0 1 0 0 0 0 1 0 0 0 1 1 0 0 0 0 0 0 1 1 0 0 0 0 1 1 0 0 0 0 0 1 0 0 0 0 1 0 0 0 0 0 0 1 0 1 1 1 1 0 0 1 1 1 1 0 0 0 1 0 0 0 0 0 1 1 0 1 1 1 0 1 0 1 1 0 0 0 1 0 1 1 0 0 0 1 0 1 0 1 1 0 1 1 1 1 1 1 0 1 0 1 1 0 1 0 0 1 1 1 0 0 0 0 0 1 1 1 0 0 1 0 1 0 1 0 1 1 0 0 1 1 1 0 0 0 1 1 0 0 1 1 1 0 1 1 1 1 1 0 1 1 1 1 0 1 0 0 0 1 1 1 1 1 1 0 1 0 0 1 1 0 1 1 1 0 0 0 0 0 1 1 0 0 1 0 1 1 0 1 0 1 0 0 1 0 0 1 1 0 1 1 0 1 0 0 1 1 1 0 1 0 1 1 1 1 0 1 0 0 0 1 1 1 0 1 1 0 1 1 0 0 0 0 1 1 1 1 0 1 1 0 1 0 1 1 1 1 1 0 0 1 1 0 1 1 1 1 0 1 1 0 0 1 1 1 1 0 1 0 0 1 0 1 1 0 1 0 0 1 1 0 1 0 0 0 0 1 1 1 0 0 0 0 0 1 1 0 0 1 0 0 0 0 1 1 0 1 0 0 1 1 0 0 1 1 0 0 0 0 0 0 0 1 0 0 0 1 0 1 1 1 1 0 1 0 0 0 1 1 1 1 1 1 0 0 1 1 1 0 1 1 1 0 0 0 0 1 0 1 1 1 1 1 1 1 0 1 1 1 0 0 1 1 1 0 1 0 1 1 1 1 0 1 0 1 1 1 1 0 1 0 1 1 1 1 0 1 1 0 1 0 1 0 1 0 0 0 1 1 0 0 0 1 0 1 0 1 0 1 1 1 1 1 0 1 0 0 1 0 0 1 1 1 1 1 1 0 0 0 1 1 0 1 1 0 0 0 1 0 1 1 1 0 1 1 0 0 1 1 1 0 1 0 0 0 1 1 1 0 0 0 1 0 0 0 0 0 0 1 0 1 0 0 0 1 1 0 0 0 0 1 0 1 1 1 1 0 1 0 0 1 0 1 0 0 0 0 0 1 0 0 1 0 0 0 1 0 1 0 1 1 0 1 0 0 0 0 1 0 1 1 1 1 1 0 1 0 0 1 0 0 1 1 0 0 1 1 1 1 1 1 1 0 0 1 1 0 0 0 1 0 1 1 0 0 0 0 0 0 0
1514 1871
1366 1369
1286 1729
1507 1817
1106 1533
