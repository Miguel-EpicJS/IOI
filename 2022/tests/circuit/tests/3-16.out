4faa78c22ee35083e49b65601698a1bd98a87a0f
OK
588565178
31154578
152598054
815496242
903968456
